src/include/miopen/generic_search.hpp

```diff
@@ -104,6 +104,12 @@ class ComputedIterator : public std::iterator<std::input_iterator_tag, Performan
     }
 
 public:
+    using iterator_category = std::input_iterator_tag;
+    using value_type        = PerformanceConfig;
+    using difference_type   = int;
+    using pointer           = void;
+    using reference         = void;
```

Comment:
Maybe
```suggestion
    using pointer           = PerformanceConfig*;
    using reference         = PerformanceConfig&;
```
or we do not need these types?

Reply:
I think they are not used thus added for portability only. @DrizztDoUrden could you double check? Thanks!

Reply:
I'd set it to meaningful values even if they are not used, just to reduce confusion and to make it more fool-proof.

Reply:
@DrizztDoUrden 
> more fool-proof

# 🤔